sonar-slang-plugin/src/test/java/com/sonarsource/slang/kotlin/KotlinConverterTest.java

```diff
@@ -340,6 +346,48 @@ public void testComplexMatchExpression() {
     assertTree(emptyWhen).isNotEquivalentTo(kotlinStatement("when (x) {}"));
   }
 
+  @Test
+  public void testForLoop() {
+    Tree kotlinStatement = kotlinStatement("for (item : Int in ints) { x = item; x = x + 1; }");
+    assertTree(kotlinStatement).isInstanceOf(LoopTree.class);
+    LoopTree forLoop = (LoopTree) kotlinStatement;
+    assertTree(forLoop.condition()).isInstanceOf(NativeTree.class);
+    assertThat(forLoop.condition().children()).hasSize(2);
+    assertTree(forLoop.condition().children().get(0)).hasParameterName("item");
+    assertTree(forLoop.condition().children().get(1)).isIdentifier("ints");
+    assertTree(forLoop.body()).isBlock(AssignmentExpressionTree.class, AssignmentExpressionTree.class);
+    assertThat(forLoop.kind()).isEqualTo(FOR);
+    assertThat(forLoop.keyword().text()).isEqualTo("for");
+    assertTree(forLoop).isNotEquivalentTo(kotlinStatement("for (item : String in ints) { x = item; x = x + 1; }"));
+    assertTree(forLoop).isNotEquivalentTo(kotlinStatement("for (it : Int in ints) { x = item; x = x + 1; }"));
+    assertTree(forLoop).isNotEquivalentTo(kotlinStatement("for (item : Int in floats) { x = item; x = x + 1; }"));
```

Comment:
Maybe we could at least one assertion where 2 trees are equivalent. Same thing for the other test methods on loops.

Reply:
Fixed in d2aca51. I have also added some comparison between kotlin and slang statements.